packages/orama/src/trees/avl.ts

```diff
@@ -172,20 +214,17 @@ export function lessThan<K, V>(node: Node<K, V>, key: K, inclusive = false): V {
   return result as V
 }
 
-function getNodeByKey<K, V>(node: Node<K, V>, key: K): Node<K, V> | null {
-  if (!node) {
-    return null
-  }
-
-  if (node.key === key) {
-    return node
-  }
-
-  if (key < node.key) {
-    return node.left ? getNodeByKey(node.left, key) : null
+function getNodeByKey<K, V>(node: Node<K, V> | null, key: K): Node<K, V> | null {
+  while (node !== null) {
+    if (key < node.key) {
+      node = node.left;
+    } else if (key > node.key) {
+      node = node.right;
+    } else {
+      return node;
+    }
   }
-
-  return node.right ? getNodeByKey(node.right, key) : null
+  return null;
```

Comment:
This is not idomatic of javascript, unless a specific case its better to just return